From 75bce4d50c4be592c6ff60b9f0d2c68200b032a2 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 19 Nov 2012 16:17:48 +0100
Subject: [PATCH 487/628] usb: ehci-xilinx-usbps: Clean Zedboard USB OTG

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 9fe9c3c97b1585feb4226f5aa8fdedc697e8780f

Remove one if condition which is always fulfil.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Acked-by: John Linn <john.linn@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/usb/host/ehci-xilinx-usbps.c |   12 +++++-------
 1 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/usb/host/ehci-xilinx-usbps.c b/drivers/usb/host/ehci-xilinx-usbps.c
index 1a0ef68..71417d6 100644
--- a/drivers/usb/host/ehci-xilinx-usbps.c
+++ b/drivers/usb/host/ehci-xilinx-usbps.c
@@ -191,13 +191,11 @@ static int usb_hcd_xusbps_probe(const struct hc_driver *driver,
 	} else {
 #ifdef CONFIG_XILINX_ZED_USB_OTG
 		pr_info ("%s: No OTG assigned!\n", __func__);
-		if (!pdata->otg) {
-			pdata->otg = otg_ulpi_create (&ulpi_viewport_access_ops,
-				ULPI_OTG_DRVVBUS | ULPI_OTG_DRVVBUS_EXT);
-			if (pdata->otg) {
-				pdata->otg->io_priv = hcd->regs + XUSBPS_SOC_USB_ULPIVP;
-				ehci->ulpi = pdata->otg;
-			}
+		pdata->otg = otg_ulpi_create(&ulpi_viewport_access_ops,
+			ULPI_OTG_DRVVBUS | ULPI_OTG_DRVVBUS_EXT);
+		if (pdata->otg) {
+			pdata->otg->io_priv = hcd->regs + XUSBPS_SOC_USB_ULPIVP;
+			ehci->ulpi = pdata->otg;
 		}
 		pr_info ("%s: OTG now assigned!\n", __func__);
 #endif
-- 
1.7.5.4

